Networking/Networking/ProductImageInBackground/MediaUploadSessionManager.swift

```diff
@@ -0,0 +1,164 @@
+import Foundation
+
+public protocol MediaUploadSessionManagerDelegate: AnyObject {
+    func mediaUploadSessionManager(_ manager: MediaUploadSessionManager,
+                                   didCompleteUpload uploadID: String,
+                                   withResult result: Result<Media, Error>)
+}
+
+/// Background upload specific errors
+enum BackgroundUploadError: Error {
+    case invalidRequestBody
+    case invalidResponse
+    case decodingError
+}
+
+/// Session Manager for media upload in background
+///
+public final class MediaUploadSessionManager: NSObject {
+
+    public let backgroundSessionIdentifier: String
+    private lazy var backgroundSession: URLSession = {
+        let config = URLSessionConfiguration.background(withIdentifier: backgroundSessionIdentifier)
+        config.sharedContainerIdentifier = "group.com.automattic.woocommerce"
+        config.sessionSendsLaunchEvents = true
+        config.isDiscretionary = false
+        config.allowsCellularAccess = true
+        return URLSession(configuration: config, delegate: self, delegateQueue: nil)
+    }()
+
+    private var completionHandlers: [String: (Result<Media, Error>) -> Void] = [:]
+    private var taskResponseData: [Int: Data] = [:]
+    private var backgroundCompletionHandler: (() -> Void)?
+    weak var delegate: MediaUploadSessionManagerDelegate?
+
+    public init(sessionIdentifier: String = "com.automattic.woocommerce.background.upload") {
```

Comment:
Should we use bundle based identifiers as suggested [here](https://www.avanderlee.com/swift/urlsession-common-pitfalls-with-background-download-upload-tasks/#bundle-based-identifiers)?